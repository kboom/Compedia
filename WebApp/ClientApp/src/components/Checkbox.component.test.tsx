import renderer from "react-test-renderer";
import { Checkbox } from "./Checkbox.atom";

const noop = () => {};

// https://jestjs.io/docs/api#describeeachtablename-fn-timeout
test.each<{ checked: boolean; readOnly: boolean }>([
  { checked: true, readOnly: true },
  { checked: true, readOnly: false },
  { checked: false, readOnly: true },
  { checked: false, readOnly: false },
])("with %s", async ({ checked, readOnly }) => {
  const tree = renderer
    .create(<Checkbox checked={checked} readOnly={readOnly} onCheck={noop} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
