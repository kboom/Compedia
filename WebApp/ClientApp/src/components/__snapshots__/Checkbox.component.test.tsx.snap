// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with [object Object] 1`] = `
<label
  className="sc-gsnTZi gnfzvj"
>
  <input
    checked={true}
    className="sc-dkzDqf gSETJl"
    onChange={[Function]}
    readOnly={true}
    type="checkbox"
  />
  <i
    className="sc-hKMtZM dREUOg"
  />
</label>
`;

exports[`with [object Object] 2`] = `
<label
  className="sc-gsnTZi gnfzvj"
>
  <input
    checked={true}
    className="sc-dkzDqf gSETJl"
    onChange={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <i
    className="sc-hKMtZM dREUOg"
  />
</label>
`;

exports[`with [object Object] 3`] = `
<label
  className="sc-gsnTZi gnfzvj"
>
  <input
    checked={false}
    className="sc-dkzDqf gSETJl"
    onChange={[Function]}
    readOnly={true}
    type="checkbox"
  />
  <i
    className="sc-hKMtZM dREUOg"
  />
</label>
`;

exports[`with [object Object] 4`] = `
<label
  className="sc-gsnTZi gnfzvj"
>
  <input
    checked={false}
    className="sc-dkzDqf gSETJl"
    onChange={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <i
    className="sc-hKMtZM dREUOg"
  />
</label>
`;

exports[`with { checked: false, readOnly: false } 1`] = `
<label
  className="sc-gsDKAQ hHlMTP"
>
  <input
    checked={false}
    className="sc-dkPtRN jjGYdK"
    onChange={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <i
    className="sc-hKwDye kmDGzT"
  />
</label>
`;

exports[`with { checked: false, readOnly: true } 1`] = `
<label
  className="sc-gsDKAQ hHlMTP"
>
  <input
    checked={false}
    className="sc-dkPtRN jjGYdK"
    onChange={[Function]}
    readOnly={true}
    type="checkbox"
  />
  <i
    className="sc-hKwDye kmDGzT"
  />
</label>
`;

exports[`with { checked: true, readOnly: false } 1`] = `
<label
  className="sc-gsDKAQ hHlMTP"
>
  <input
    checked={true}
    className="sc-dkPtRN jjGYdK"
    onChange={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <i
    className="sc-hKwDye kmDGzT"
  />
</label>
`;

exports[`with { checked: true, readOnly: true } 1`] = `
<label
  className="sc-gsDKAQ hHlMTP"
>
  <input
    checked={true}
    className="sc-dkPtRN jjGYdK"
    onChange={[Function]}
    readOnly={true}
    type="checkbox"
  />
  <i
    className="sc-hKwDye kmDGzT"
  />
</label>
`;
